code/map/MapHandler.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+
+using Sandbox;
+
+using TTTReborn.Globals;
+using TTTReborn.Items;
+
+namespace TTTReborn.Map
+{
+    public partial class MapHandler
+    {
+        public TTTMapSettings MapSettings { get; private set; }
+
+        public List<ModelEntityData> ModelEntityDataList;
+
+        public MapHandler()
+        {
+            ModelEntityDataList = new();
+
+            foreach (Entity entity in Entity.All)
+            {
+                if (entity is TTTMapSettings mapSettings)
+                {
+                    MapSettings = mapSettings;
+                    MapSettings.FireSettingsSpawn();
+                }
+                else if (entity is Sandbox.Prop || entity is BaseCarriable)
+                {
+                    ModelEntityDataList.Add(ModelEntityData.Create(entity as ModelEntity));
+                }
+            }
+        }
+
+        public void Reset()
+        {
+            List<TTTAmmoRandom> randomAmmo = new();
+            List<TTTWeaponRandom> randomWeapons = new();
+
+            foreach (Entity entity in Entity.All)
+            {
+                if (entity is Sandbox.Prop || entity is BaseCarriable || entity.Tags.Has(IItem.ITEM_TAG))
+                {
+                    entity.Delete();
+                }
+                else if (entity is TTTAmmoRandom ammoRandom)
+                {
+                    randomAmmo.Add(ammoRandom); // Throws `Collection was Modified` if we activate here. Worth looking further into cleanup wise.
+                }
+                else if (entity is TTTWeaponRandom weaponRandom)
+                {
+                    randomWeapons.Add(weaponRandom); // See above comment.
+                }
+                else if (entity is TTTLogicButton button)
+                {
+                    button.Cleanup();
+                }
+
+                entity.RemoveAllDecals();
+            }
```

Comment:
So I wonder if the `Delete` of the entity causes the `collection was modified`. It's just a guess, you'd have two options here if you want to test it.

1. Iterate through the `Entity.All` array backwards (go from end to the start)
2. Add the entities you plan to delete to a new array, then delete all at the end.